vmp/resources/waf/custom_rule_set.go

```diff
@@ -0,0 +1,615 @@
+																	"Example: \\\n" +
+																	"If the type property is set to REQUEST_HEADERS and this property is set to User-Agent, then this match condition is restricted to the User-Agent request header. If the value property is omitted, then this match condition applies to all request headers.",
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+							Description: "sec_rule object describes a custom rule",
+						},
+					},
+				},
+				Description: "Contains custom rules. Each directive object defines a custom rule via the sec_rule object \\\n" +
+					"    Note: You may create up to 10 custom rules.",
+			},
+		},
+	}
+}
+
+func ResourceCustomRuleSetCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	var diags diag.Diagnostics
+
+	config := m.(**api.ClientConfig)
+
+	wafService, err := buildWAFService(**config)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	accountNumber := d.Get("account_number").(string)
+
+	log.Printf("[INFO] Creating WAF Rate Rule for Account >> %s", accountNumber)
+
+	customRuleSet := sdkwaf.CustomRule{
+		Name: d.Get("name").(string),
+	}
+
+	directive, err := ConvertInterfaceToDirectives(d.Get("directive"))
+	if err != nil {
+		return diag.Errorf("error parsing directive: %+v", err)
+	}
+
+	customRuleSet.Directives = *directive
+
+	log.Printf("[DEBUG] Name: %+v\n", customRuleSet.Name)
+	log.Printf("[DEBUG] Directive(s): %+v\n", customRuleSet.Directives)
+
+	resp, err := wafService.AddCustomRuleSet(customRuleSet, accountNumber)
+
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[INFO] %+v", resp)
+
+	d.SetId(resp.ID)
+
+	return diags
+}
+
+func ResourceCustomRuleSetRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	return diags
+}
+
+func ResourceCustomRuleSetUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	return diags
+}
+
+func ResourceCustomRuleSetDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	return diags
+}
+
+func ConvertInterfaceToDirectives(attr interface{}) (*[]sdkwaf.Directive, error) {
+
+	if set, ok := attr.(*schema.Set); ok {
+
+		items := set.List()
+
+		directives := make([]sdkwaf.Directive, 0)
+
+		for _, item := range items {
+			curr := item.(map[string]interface{})
+
+			directive := sdkwaf.Directive{}
+
+			secRule, err := ConvertInterfaceToSecRule(curr["sec_rule"])
+			if err != nil {
+				return nil, err
+			}
+
+			directive.SecRule = *secRule
+
+			directives = append(directives, directive)
+		}
+
+		return &directives, nil
+
+	} else {
+		return nil, errors.New("ConvertInterfaceToDirectives --> attr input was not a *schema.Set")
+	}
+
+}
+
+func ConvertInterfaceToSecRule(attr interface{}) (*sdkwaf.SecRule, error) {
```

Comment:
Similarly, this should be `ExpandSecRule`

Reply:
Updated.

Reply:
Updated